src/evolution_scene.c

```diff
@@ -0,0 +1,1566 @@
+    case 8:
+        sEvoGraphicsTaskId = LaunchTask_PostEvoSparklesSet1();
+        gTasks[taskId].tState++;
+        break;
+    case 9:
+        if (!gTasks[sEvoGraphicsTaskId].isActive)
+        {
+            sEvoGraphicsTaskId = LaunchTask_PostEvoSparklesSet2AndFlash_Trade(gTasks[taskId].tPostEvoSpecies);
+            gTasks[taskId].tState++;
+        }
+        break;
+    case 10:
+        if (!gTasks[sEvoGraphicsTaskId].isActive)
+        {
+            PlaySE(SE_EXP);
+            gTasks[taskId].tState++;
+        }
+        break;
+    case 11:
+        /*
+         * BUG: This check causes the evolved Pokemon's cry to play over the sfx.
+         * Negate the below condition.
+         */
+        if (IsSEPlaying())
+        {
+//            Free(sEvoMovingBgPtr);
+            PlayCry1(gTasks[taskId].tPostEvoSpecies, 0);
+            memcpy(&gPlttBufferUnfaded[0x20], sEvoStructPtr->savedPalette, 0x60);
+            gTasks[taskId].tState++;
+        }
+        break;
+    case 12:
+        if (IsCryFinished())
+        {
+            StringExpandPlaceholders(gStringVar4, gText_CongratsPkmnEvolved);
+            DrawTextOnTradeWindow(0, gStringVar4, 1);
+            PlayFanfare(MUS_FANFA5);
+            gTasks[taskId].tState++;
+            SetMonData(mon, MON_DATA_SPECIES, (&gTasks[taskId].tPostEvoSpecies));
+            CalculateMonStats(mon);
+            EvolutionRenameMon(mon, gTasks[taskId].tPreEvoSpecies, gTasks[taskId].tPostEvoSpecies);
+            GetSetPokedexFlag(SpeciesToNationalPokedexNum(gTasks[taskId].tPostEvoSpecies), FLAG_SET_SEEN);
+            GetSetPokedexFlag(SpeciesToNationalPokedexNum(gTasks[taskId].tPostEvoSpecies), FLAG_SET_CAUGHT);
+            IncrementGameStat(GAME_STAT_EVOLVED_POKEMON);
+        }
+        break;
+    case 13:
+        if (!IsTextPrinterActive(0) && IsFanfareTaskInactive() == TRUE)
+        {
+            var = MonTryLearningNewMove(mon, gTasks[taskId].tLearnsFirstMove);
+            if (var != 0 && !gTasks[taskId].tEvoWasStopped)
+            {
+                u8 text[20];
+
+                gTasks[taskId].tBits |= TASK_BIT_LEARN_MOVE;
+                gTasks[taskId].tLearnsFirstMove = FALSE;
+                gTasks[taskId].tLearnMoveState = 0;
+                GetMonData(mon, MON_DATA_NICKNAME, text);
+                StringCopy10(gBattleTextBuff1, text);
+
+                if (var == MON_HAS_MAX_MOVES)
+                    gTasks[taskId].tState = 20;
+                else if (var == MON_ALREADY_KNOWS_MOVE)
+                    break;
+                else
+                    gTasks[taskId].tState = 18;
+            }
+            else
+            {
+                PlayBGM(MUS_SHINKA);
+                DrawTextOnTradeWindow(0, gText_CommunicationStandby5, 1);
+                gTasks[taskId].tState++;
+            }
+        }
+        break;
+    case 14:
+        if (!IsTextPrinterActive(0))
+        {
+            DestroyTask(taskId);
+            Free(sEvoStructPtr);
```

Comment:
use `FREE_AND_SET_NULL` macro

Reply:
I'm neither using nor enforcing that macro.